### DISCUSSION TITLE
i18n undefined var locales

### DISCUSSION THREAD

**Discussion by FAITH-ORAN** [vercel/next.js#44429] [Help] [answered]
Hello to everyone , 
i use i18n for two différents project, in my personal project it's work fine, but in the project of my team, i got undefined :ServerRouter {
  route: '/',
  pathname: '/',
  query: {},
  asPath: '/',
  isFallback: false,
  basePath: '',
  locale: undefined,
  locales: undefined,
  defaultLocale: undefined,
  isReady: false,
  domainLocales: undefined,
  isPreview: false,
  isLocaleDomain: false
}
=> in may page next.config.cjs i have:
/*eslint linebreak-style: ["error", "windows"]*/
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  i18n:{
    locales: ["en-US", "fr", "nl-NL"],
    defaultLocale: "fr",
  },
}

module.exports = nextConfig

and in index.js:
/*eslint linebreak-style: ["error", "windows"]*/
import { useRouter } from "next/router"
import Head from "next/head"
import Image from "next/image"
import styles from "../styles/Home.module.css"

import Link from "next/link"

export default function Home() {
  const router = useRouter()
  const {locale, locales, push}=useRouter()
   console.log(router)
return (

 <div>
  <h1> {locale} </h1> 
 <div/>
)
}

can you help me please, thank you

**Comment by FAITH-ORAN** [comment]
in return it's a div wrap h1 inside of it the locale var

**Comment by raulfdm** [comment, marked as answer]
> Don't get this wrong; I'm trying to help you to get help

It's really hard to help you out here. That's because you post some code snippets without too much context, and the last one is missing the rest of the content.

Here is a suggestion I can give so you can easily get help from the community:

1. whenever you post code snippets, scope them into Markdown code blocks with file extension ([see how](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks));
2. a reproducible and standalone project is easier to understand. For that, you could either:
    - Give us a link to your project if the setup is easy;
    - Create another repository with a minimal code to reproduce the error;
    - (best option) Create a stackblitz NextJS project (https://stackblitz.com/), make the error happen, and share the link
3. Organize your help into blocks (e.g. "## Description", "## Problem", "## More info", "## Expected behavior"). It's easier for people who's reading

I bet if you follow these tips you'll the help you need.

I wish you good luck

**Reply by FAITH-ORAN** [reply]
Thank you for your message, i resolve the probleme , first of all: i hade a file=> next.config.cjs i change the extension into mjs file. and i delete from my package.Json => type:module

**Reply by raulfdm** [reply]
nice... you can mark it as "answered" 😄
